chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala

```diff
@@ -1,30 +1,92 @@
 package org.bitcoins.chain.util
 
-import org.bitcoins.chain.models.BlockHeaderDb
-import org.bitcoins.core.protocol.blockchain.RegTestNetChainParams
+import org.bitcoins.chain.api.ChainApi
+import org.bitcoins.chain.blockchain.{Blockchain, ChainHandler}
+import org.bitcoins.chain.db.ChainDbManagement
+import org.bitcoins.chain.models.{BlockHeaderDAO, BlockHeaderDb}
+import org.bitcoins.core.protocol.blockchain.{
+  ChainParams,
+  RegTestNetChainParams
+}
+import org.bitcoins.core.util.BitcoinSLogger
 import org.bitcoins.db.{DbConfig, UnitTestDbConfig}
 import org.bitcoins.testkit.chain.ChainTestUtil
-import org.scalatest.{
-  AsyncFlatSpec,
-  BeforeAndAfter,
-  BeforeAndAfterAll,
-  MustMatchers
-}
+import org.scalatest._
 
-import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
 
 trait ChainUnitTest
     extends AsyncFlatSpec
     with MustMatchers
+    with BitcoinSLogger
     with BeforeAndAfter
     with BeforeAndAfterAll {
 
   val timeout = 10.seconds
   def dbConfig: DbConfig = UnitTestDbConfig
-  val genesisHeader: BlockHeaderDb = ChainTestUtil.regTestGenesisHeaderDb
-  val networkParam: RegTestNetChainParams.type = RegTestNetChainParams
+
+  val genesisHeaderDb: BlockHeaderDb = ChainTestUtil.regTestGenesisHeaderDb
+  val chainParam: ChainParams = RegTestNetChainParams
+
+  lazy val blockHeaderDAO = BlockHeaderDAO(chainParams =
+                                             ChainTestUtil.regTestChainParams,
+                                           dbConfig = dbConfig)
+
+  lazy val blockchain =
+    Blockchain.fromHeaders(Vector(genesisHeaderDb), blockHeaderDAO)
+
+  lazy val chainHandler: ChainHandler = ChainHandler(blockchain)
+  lazy val chainApi: ChainApi = chainHandler
 
   implicit def ec: ExecutionContext =
     scala.concurrent.ExecutionContext.Implicits.global
+
+  /** Fixture that creates a [[org.bitcoins.chain.models.BlockHeaderTable]]
+    * with one row inserted into it, the [[RegTestNetChainParams]]
+    * genesis block
+    * @param test
+    * @return
+    */
+  def withBlockHeaderDAO(
+      test: BlockHeaderDAO => Future[Assertion]): Future[Assertion] = {
+    val tableSetupF = ChainDbManagement.createHeaderTable(dbConfig = dbConfig,
```

Comment:
We can't have multiple tests running with this at the same time right?

Reply:
This is actually async safe i believe because i moved the database config to be in memory https://github.com/bitcoin-s/bitcoin-s-core/pull/398/files#diff-66e7e260c6c9a87ddf216be43e4fe223R37